internal/api/impl/v1/dashboard/validator/validator.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validator
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"cuelang.org/go/cue"
+	"cuelang.org/go/cue/cuecontext"
+	"cuelang.org/go/cue/load"
+	"github.com/perses/perses/internal/config"
+	v1 "github.com/perses/perses/pkg/model/api/v1"
+	"github.com/sirupsen/logrus"
+)
+
+/*
+ * A Validator can be used to run checks on Dashboard objects, based on cuelang definitions
+ */
+type Validator interface {
+	Process(dashboard *v1.Dashboard) error
+	LoadSchemas(path string)
+}
+
+type validator struct {
+	schemasPath string
+	context     *cue.Context
+	baseDef     cue.Value
+	schemas     *sync.Map
+}
+
+const (
+	baseDefFileName   = "base.cue"
+	pluginsFolderName = "plugins"
+)
+
+/*
+ * Instanciate a validator
+ */
+func New(conf config.Schemas) Validator {
+	return &validator{
+		schemasPath: conf.Path,
+		context:     cuecontext.New(),
+		baseDef:     cue.Value{},
+		schemas:     &sync.Map{},
+	}
+}
+
+/*
+ * Validate a dashboard.
+ * The dashboard's panels are matched against the known list of CUE definitions (schemas).
+ * If no schema matches, the validation fails.
+ */
+func (v *validator) Process(dashboard *v1.Dashboard) error {
```

Comment:
I'm wondering if it wouldn't be nicer to have a list of panels here instead: 

```go
func (v *validator) Validate(panels map[string]json.RawMessage) error {
}
```
What do you think ?

Reply:
Ah yes I forgot to note it when we discussed it :x 'will do